boards/Kconfig

```diff
@@ -3118,6 +3118,14 @@ config BOARDCTL_UNIQUEID_SIZE
 		Provides the size of the memory buffer that must be provided by the
 		caller of board_uniqueid() in which to receive the board unique ID.
 
+config BOARDCTL_NETCONF
+	bool "Return board network configuration"
+	default n
+	---help---
+		Enables support for the BOARDIOC_NETCONF boardctl() command.
+		Architecture specific logic must provide the board_get_netconf()
+		interface.
+
```

Comment:
Wouldn't the procfs be a better, more standard way to get configuration information?  boardctl() is a necessary kludge, but we should really use it as little as possible since it is inherently non-POSIX, non-standard.  procfs is the recommended way.

The preferred way would be to support a procfs folder with files named eth0, eth1, etc.  Reading any one of this files would return the configuration of that network.

Is there any precedence for anything like this in Linux?

Reply:
We should add it. But It is not appropriate for a small constrained system.

Reply:
> We should add it. But It is not appropriate for a small constrained system.

Irrelevant.  Let's do what is right for the OS!  Commonality of interances, standards, compatibility.  THAT is what is important.